service/create/service.go

```diff
@@ -1003,31 +1025,28 @@ func (s *Service) onDelete(obj interface{}) {
 		s.logger.Log("info", "deleted masters")
 	}
 
-	// Delete workers.
-	s.logger.Log("info", "deleting workers...")
-	if err := s.deleteMachines(deleteMachinesInput{
-		clients:     clients,
-		clusterName: cluster.Name,
-		prefix:      prefixWorker,
-	}); err != nil {
+	// Delete workers Auto Scaling Group.
+	asg := awsresources.AutoScalingGroup{
+		Client: clients.AutoScaling,
+		Name:   cluster.Name,
+	}
+
+	if err := asg.Delete(); err != nil {
 		s.logger.Log("error", errgo.Details(err))
 	} else {
-		s.logger.Log("info", "deleted workers")
+		s.logger.Log("info", "deleted workers auto scaling group")
 	}
 
-	// Delete worker launch configuration.
-	workerLCName, err := launchConfigurationName(cluster, prefixWorker)
-	if err != nil {
+	// Delete workers launch configuration.
+	lcInput := launchConfigurationInput{
+		clients: clients,
+		cluster: cluster,
+		prefix:  "worker",
```

Comment:
Re my comment above: here would be only: `name: launchConfigurationName(cluster, "worker", workersSecurityGroupID)`